src/projects/projectsRestApiClient.ts

```diff
@@ -0,0 +1,45 @@
+import { Axios } from 'axios';
+import ProjectsApiClient from './projectsApiClient';
+import { getBaseRequestConfig } from '../util/http';
+import Constants from '../constants';
+import Project from './project';
+
+/** Implementation of ProjectsApiClient that goes through the REST API. */
+export default class ProjectsRestApiClient extends ProjectsApiClient {
+  /** Axios client to send all HTTP requests. */
+  client: Axios;
+
+  /** Session used for authentication. Used only for Cloud host. */
+  session: string | undefined;
+
+  /**
+   * Constructor for Projects API client.
+   * @param {Axios} client - Axios instance to send all HTTP requests.
+   */
+  constructor(client: Axios) {
+    super();
+    this.client = client;
```

Comment:
It's looks prety nice, but thinking are you consider in the future to use enum or avoid using  "this" making encapsulation or creating "type"  for specify cases?